### PR TITLE
ibrohim

### PR DESCRIPTION
**Pull request by ibrohim232**

### CODE REVIEW

File: src/main/java/com/example/iticket/ITicketApplication.java [hunk 1]
```diff
@@ -8,6 +8,7 @@ public class ITicketApplication
 {
 
     public static void main( String[] args )
+            //just
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/ITicketApplication.java:11]: kommentni o'chiring yoki yaxshiroq komment qo'ying

File: src/main/java/com/example/iticket/block/BlockController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.util.UUID;
+
+@Controller(value = "/block")
+@RequiredArgsConstructor
+public class BlockController {
+    private final BlockService blockService;
+    private final BlockDTOMapper blockDTOMapper;
+
+    @GetMapping("/id")
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockController.java:16]: path variable ishlating. shunchaki id stringni emas

File: src/main/java/com/example/iticket/block/BlockController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.example.iticket.block;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@Controller(value = "/block")
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockController.java:10]: restControllernin ishlating

File: src/main/java/com/example/iticket/block/BlockController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    private final BlockService blockService;
+    private final BlockDTOMapper blockDTOMapper;
+
+    @GetMapping("/id")
+    public BlockDTO get(@RequestParam("id") UUID id) {
+        return blockDTOMapper.apply(blockService.get(id));
+    }
+
+    @PostMapping
+    public void add(@RequestBody BlockDTO blockDTO) {
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockController.java:22]: @valid annotation ni qo'ying. not null validation ishlashi uchun

File: src/main/java/com/example/iticket/block/BlockController.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.UUID;
+
+@Controller(value = "/block")
+@RequiredArgsConstructor
+public class BlockController {
+    private final BlockService blockService;
+    private final BlockDTOMapper blockDTOMapper;
+
+    @GetMapping("/id")
+    public BlockDTO get(@RequestParam("id") UUID id) {
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockController.java:17]: request param da id ni berilmidi

File: src/main/java/com/example/iticket/block/BlockController.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        blockService.add(block);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable UUID id) {
+        blockService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    public void update(@PathVariable("id") UUID id, @RequestBody BlockDTO blockDTO) {
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockController.java:33]: @valid annotation

File: src/main/java/com/example/iticket/block/BlockDTOMapper.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    @Override
+    public BlockDTO apply(Block block) {
+        return new BlockDTO(
+                block.getHallId(),
+                block.getName(),
+                block.getCountOfSeat()
+        );
+    }
+
+    public Block apply(BlockDTO blockDTO) {
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockDTOMapper.java:20]: Methodni ismini to'gri qo'ying. masalan toEntity

File: src/main/java/com/example/iticket/block/BlockDTOMapper.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+import java.util.function.Function;
+
+@Service
+public class BlockDTOMapper implements Function<Block, BlockDTO> {
+    @Override
+    public BlockDTO apply(Block block) {
```
Comment (aabdurashidov) [inline src/main/java/com/example/iticket/block/BlockDTOMapper.java:12]: methodni ismini o'zgartiring. masalan toResponseDTO

**Review by aabdurashidov — CHANGES_REQUESTED** [review]

**Review by aabdurashidov — COMMENTED** [review]
Aytganday. Liquibase ham ishlating !!!!

File: src/main/java/com/example/iticket/ITicketApplication.java (hunk 1 above)
Comment (ibrohim232) [reply src/main/java/com/example/iticket/ITicketApplication.java:11]: ochirib tashadim

File: src/main/java/com/example/iticket/block/BlockDTOMapper.java (hunk 8 above)
Comment (ibrohim232) [reply src/main/java/com/example/iticket/block/BlockDTOMapper.java:12]: buni implement qiganida shu  nom bilan edi

**Pull request closed without merging**